File: static/CWE-78/python/vulnerable.py
# CWE-78: OS Command Injection
# User-controlled request data is passed into a shell command.
# Vulnerable sink: subprocess.call(cmd, shell=True)

import subprocess
from flask import Flask, request

app = Flask(__name__)


@app.route("/ping")
def ping():
    host = request.args.get("host")
    cmd = "ping -c 1 " + host
    subprocess.call(cmd, shell=True)
    return "pinged " + host

File: static/CWE-134/c/vulnerable.c
/* CWE-134: Use of Externally-Controlled Format String
   User-controlled argv is passed directly as the format string.
   Vulnerable sink: printf(argv[1]) */
#include <stdio.h>

int main(int argc, char **argv)
{
    if (argc < 2)
        return 1;

    /* Attacker controls the format string: %n, %x, etc. */
    printf(argv[1]);
    printf("\n");

    return 0;
}
